xl2times/transforms.py

```diff
@@ -1033,33 +971,35 @@ def remove_invalid_values(
         Tag.uc_t,
     }
 
-    def remove_table_invalid_values(
-        table: EmbeddedXlTable,
-    ) -> EmbeddedXlTable:
-        """Remove invalid entries in a table dataframe."""
-        df = table.dataframe.copy()
+    def remove_invalid_values_table(df):
         is_valid_list = [
             df[colname].isin(values)
             for colname, values in constraints.items()
             if colname in df.columns
         ]
         if is_valid_list:
             is_valid = reduce(lambda a, b: a & b, is_valid_list)
-            df = df[is_valid]
-            df.reset_index(drop=True, inplace=True)
-        table = replace(table, dataframe=df)
-        return table
+            return df[is_valid].reset_index(drop=True)
+        else:
+            return df
 
-    return [
-        remove_table_invalid_values(t) if t.tag not in skip_tags else t for t in tables
-    ]
+    for tag, dataframe in tables.items():
+        if tag not in skip_tags:
+            df = dataframe.copy()
+            tables[tag] = remove_invalid_values_table(df)
```

Comment:
```suggestion
            tables[tag] = remove_invalid_values_table(dataframe)
```
I think we don't need to copy the dataframe, since we don't use the old ones anymore?